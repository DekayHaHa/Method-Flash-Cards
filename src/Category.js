import React, { Component } from 'react';
import Snippet from './Snippet'

export default class Categories extends Component {
	constructor(props) {
		super(props);
		this.state = {
			questions: [],
			display: false,
			currentQuestion: {}
		}
	}
	componentWillMount(){
		this.setState({questions: this.props.questions})
	}
	getQuestion = () => {
		const prompts = this.state.questions
		const question = prompts.splice(Math.floor(Math.random() * prompts.length), 1)
		this.setState({questions: prompts, currentQuestion: question[0], display: true})
	}
	displayToggle = () => {
		this.setState({display: false})
	}
	render() {
		const {display, questions} = this.state
		let buttonDisable = display || questions.length === 0 ? true : false;
		return (
			<div>
				<button type='button' disabled={buttonDisable} onClick={this.getQuestion}>{this.props.title}</button>
				{this.state.display && <Snippet question={this.state.currentQuestion} close={this.displayToggle} keepScore={this.props.keepScore}/>}
			</div>
		);
	}
}