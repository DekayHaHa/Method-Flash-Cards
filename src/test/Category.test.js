import React from 'react';
import ReactDOM from 'react-dom';
import Category from '../Category';
import { shallow } from 'enzyme';

const resetMock = jest.fn();
const mockCards = [{
	"snippet": ["let arr = [1, 2, 3, 4, 5]", "let results = arr.", "(val => val > 0)", "console.log(results) // logs true"],
	"category": "Array Prototype Method",
	"solution": "every",
	"mdn": "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every",
	"uid": "card-1"
},
	{
		"snippet": ["let arr = [1, 2, 3, 4, 5]", "let results = arr.", "(5)", "console.log(results) // logs 4"],
		"category": "Array Prototype Method",
		"solution": "indexOf",
		"mdn": "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/indexOf",
		"uid": "card-2"
	},
	{
		"snippet": ["let arr = [1, 2, 3, 4, 5]", "let results = arr.", "(val => val > 3)", "console.log(results) // logs [ 4, 5 ]"],
		"category": "Array Prototype Method",
		"solution": "filter",
		"mdn": "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter",
		"uid": "card-3"
	}]

describe('Category', () => {
	let wrapper;
	beforeEach(() => {
		wrapper = shallow(
			<Category resetToggle={resetMock} questions={mockCards}/>
		);
	});
	it('should match snapshop', () => {
		expect(wrapper).toMatchSnapshot();
	});
	it('should have default state', () => {
		expect(wrapper.state()).toEqual({
			questions: mockCards,
			display: false,
			currentQuestion: {}
		});
	});
	it('should toggle snippet display', () => {
		wrapper.instance().displayToggle();
		expect(wrapper).toMatchSnapshot();
	})
	it('should grab question for current question after click', () => {
		expect(wrapper.state('currentQuestion')).toEqual({})
		wrapper.find('button').simulate('click')
		expect(wrapper.state('currentQuestion')).toHaveProperty('snippet')
		expect(wrapper.state('display')).toEqual(true)
	})
})