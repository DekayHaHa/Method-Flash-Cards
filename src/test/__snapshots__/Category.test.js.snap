// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category should match snapshop 1`] = `
<div
  className="cat-container"
>
  <button
    className="cat-button"
    disabled={false}
    onClick={[Function]}
  />
</div>
`;

exports[`Category should toggle snippet display 1`] = `
<div
  className="cat-container"
>
  <button
    className="cat-button"
    disabled={false}
    onClick={[Function]}
  />
</div>
`;
